### PR TITLE
Fix `istioctl x workload entry configure` when no proxyMetadata is sp…

### PR DESCRIPTION
**Pull request by kevinnoel-be**
…ecified.

Created a separate unit test as unmarshalling from a file doesn't produce the error.

Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[X] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[X] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[X] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Review by stevenctl — APPROVED** [review]
Thanks for the PR!

File: istioctl/cmd/workload_test.go [hunk 1]
```diff
@@ -248,3 +279,30 @@ func runTestCmd(t *testing.T, args []string) (string, error) {
...
+		if checkGolden {
+			t.Run(f.Name(), func(t *testing.T) {
+				contents := util.ReadFile(path.Join(testdir, f.Name()), t)
+				goldenFile := path.Join(testdir, f.Name()+goldenSuffix)
+				util.RefreshGoldenFile(contents, goldenFile, t)
+				util.CompareContent(contents, goldenFile, t)
+			})
+		}
+	}
+}
```
Comment (shamsher31) [inline istioctl/cmd/workload_test.go:308]: nit:
```suggestion
}

```

**Review by howardjohn — APPROVED** [review]

**Comment by kevinnoel-be** [comment]
> LGTM, please fix the `make gen` issue.

@shamsher31 Thanks for the review, I've fixed the `make gen` issue.

**Review by shamsher31 — APPROVED** [review]
Just one more lint issue to fix. run `make lint` and commit your changes.

**Comment by kevinnoel-be** [comment]
@shamsher31 Fixed, thanks again. Will this fix be backported to previous releases?

**Comment by shamsher31** [comment]
> Will this fix be backported to previous releases?

I think so. But @howardjohn can confirm.

**Pull request merged**